rewrite-gradle/src/main/java/org/openrewrite/gradle/DependencyVersionSelector.java

```diff
@@ -188,8 +191,10 @@ private List<MavenRepository> determineRepos(@Nullable String configuration) {
         if (gradleProject == null) {
             throw new IllegalStateException("Gradle project must be set to determine repositories."); // Caught by caller
         }
-        return "classpath".equals(configuration) ?
-                gradleProject.getBuildscript().getMavenRepositories() :
-                gradleProject.getMavenRepositories();
+        if ("classpath".equals(configuration)) {
+            return gradleProject.getBuildscript().getMavenRepositories();
+        }
+        List<MavenRepository> repos = gradleProject.getMavenRepositories();
+        return repos.isEmpty() ? gradleProject.getBuildscript().getMavenRepositories() : repos;
```

Comment:
The logic here is invalid from a Gradle standpoint.

Buildscript should be used for plugins and the `classpath` configuration while project repositories should be used for project configurations. The logic here that does the fallback to the buildscript repositories when project repositories is empty does not happen internally in Gradle, so I would recommend not doing it here either.

Reply:
⬇️